### DISCUSSION TITLE
Ratelimting for POST?

### DISCUSSION THREAD

**Discussion by Alee14** [supabase/supabase#21152] [Questions]
Hi,
I am making a guestbook for my website, and was wondering if there is a way to put a "ratelimit" for POSTing? Like people can send a guestbook every hour per ip or user device.

**Comment by GaryAustin1** [comment]
So these are anon requests or authenticated users?

You could use an rpc call to a security definer function and check the table you are going to insert to for a post by the user identifier in the past hour.  If not then do the post with the identifier in a column.  

You can get the ip info from a header in the function, or pass in some device info you make up to the rpc call.
Info on getting headers in Postgres: https://postgrest.org/en/stable/references/transactions.html#request-headers-cookies-and-jwt-claims

This is not a security approach as the user can always fake things, but stops casual over posting by a user.

**Reply by Alee14** [reply]
> So these are anon requests or authenticated users?
> 
> You could use an rpc call to a security definer function and check the table you are going to insert to for a post by the user identifier in the past hour. If not then do the post with the identifier in a column.
> 
> You can get the ip info from a header in the function, or pass in some device info you make up to the rpc call. Info on getting headers in Postgres: https://postgrest.org/en/stable/references/transactions.html#request-headers-cookies-and-jwt-claims
> 
> This is not a security approach as the user can always fake things, but stops casual over posting by a user.

Forgot to mention it but, anon requests

**Reply by GaryAustin1** [reply]
Yeah then the ip or you sending a device id would be the possible ways.  As I said though these are just to stop casual users not a determined attacker who fakes those randomly.  Obviously the ip header takes more work to fake as it would require a server I believe.
